docs/how_pool_works.md

```diff
@@ -0,0 +1,189 @@
+as a receiver with weight 1, but only half as big as another receiver with weight 4.
+
+## Starting sending
+
+When the sender has a sufficient balance, a sending rate and a list of receivers, sending starts.
+First, the funding period is calculated.
+Its start is the current block and its end is the block on which the balance will run out.
+Next, for each receiver the weighted share of the per-block rate is calculated.
+The receiver's deltas are updated to reflect that during the whole sending period on every block
+it's going to receive the calculated amount.
+
+Let's take a look at an example of an application of a delta.
+The sender will be sending 1 per block or 5 per cycle.
+
+![](how_pool_works_4.png)
+
+The deltas are applied relative to the existing values.
+It doesn't matter if anybody else is funding the receiver, it won't affect this sender.
+
+Another important point is that the delta changes are usually split between two cycles.
+This reflects that the first cycle is only partially affected by the change in funding.
+Only the second one is fully affected and it must apply the rest of the delta.
+
+In this case, the total change of the per-cycle delta is +5 to start sending.
+The current cycle isn't fully affected though, only 2 out of 5 blocks are sending.
+It's effectively going to transfer only the amount of 2, which is reflected in the +2 delta change.
+On the other hand, the next cycle and the ones after it are going to transfer the full 5.
+This is expressed with the +3 delta change, which turns 2 per cycle into the full 5 per cycle.
+
+## Stopping sending
+
+When funding stops, the deltas need to be reverted.
+To do that basically the same process is applied, just with negative deltas.
+
+In this case, the reverting is split into 2 cycles too, one with -4 and the other with -1.
+
+Let's assume that a few blocks have passed, but the sender wants to stop sending.
+This can happen because the sender doesn't want to fund the receiver anymore
+or because they want to change some of its configuration.
+In the latter case sending is stopped only to be immediately resumed, but with different parameters.
+Either way, the effects of the sender on the receiver's deltas need to be reverted
+from the current block to the end of the existing funding period.
+
+![](how_pool_works_5.png)
+
+The old funding end deltas are reverted because they don't reflect the real funding end anymore.
+On the other hand, a new end is applied to the current block,
+just as if it was always supposed to be the end of the funding period.
+Now the receiver's future isn't affected by the sender anymore.
+The past stays untouched because the already sent funds are out of the sender's control.
+
+# The proxy
+
+The proxy is configured only with a list of weighted receivers.
+It's not possible to send funds via a proxy which isn't configured.
+The sum of the receivers' weights must always be a constant value,
```

Comment:
> weights must always be a constant value,

under which context? Not sure I understand? If I start with `[(a, 1), (b, 1)]` and then change it to `[(a, 1), (b, 2)]`, the sum of the weights is no longer the same as it was before. That's how I mean my question :)

Reply:
You're right, if you try to make such change, you break the rule about the sum of weights, which is described here. There's no context, it's a global rule. What's your question?